### PR TITLE
Issue27

### LINKED ISSUE #27: Implement SchoolController.addStudent(Student)
This should call a method on the Database, for example save(Student), but before calling save on the database it should first check whether a student with the same id already exists in the database. If it exists it should return false, otherwise it should return true

### PR DESCRIPTION
**Pull request by Michedev**
Close #27 
Implement SchoolController.addToDB(Student) method and Database.add(Student)

### CODE REVIEW

File: com.examples.exercise/src/test/java/com/examples/exercise/SchoolControllerTest.java [hunk 1]
```diff
@@ -48,15 +48,39 @@ public void testGetStudentByIdWithBadIndex() {
...
 		schoolController.getAllStudents();
 	}
+	
+	@Test
+	public void testAddStudentBase(){
+		Student student = factoryStudent("1", "Pippo");
+		when(database.exists(anyString())).thenReturn(false);
+		assertTrue(schoolController.addToDB(student));
+		verify(database, times(1)).add(student);
+
```
Comment (LorenzoBettini) [inline com.examples.exercise/src/test/java/com/examples/exercise/SchoolControllerTest.java:85]: Please remove this empty line

File: com.examples.exercise/src/test/java/com/examples/exercise/SchoolControllerTest.java [hunk 2]
```diff
@@ -48,15 +48,39 @@ public void testGetStudentByIdWithBadIndex() {
...
+	}
+	
+	@Test
+	public void testAddStudentWhenIsDuplicate(){
+		Student student = factoryStudent("1", "Pippo");
+		when(database.exists(anyString())).thenReturn(true);
+		assertFalse(schoolController.addToDB(student));
+		verify(database, times(1)).exists(student.getId());
+
```
Comment (LorenzoBettini) [inline com.examples.exercise/src/test/java/com/examples/exercise/SchoolControllerTest.java:95]: Please remove this empty line

File: com.examples.exercise/src/test/java/com/examples/exercise/SchoolControllerTest.java [hunk 3]
```diff
@@ -48,15 +48,39 @@ public void testGetStudentByIdWithBadIndex() {
...
+		verify(database, times(1)).add(student);
+
+	}
+	
+	@Test
+	public void testAddStudentWhenIsDuplicate(){
+		Student student = factoryStudent("1", "Pippo");
+		when(database.exists(anyString())).thenReturn(true);
+		assertFalse(schoolController.addToDB(student));
+		verify(database, times(1)).exists(student.getId());
```
Comment (LorenzoBettini) [inline com.examples.exercise/src/test/java/com/examples/exercise/SchoolControllerTest.java:93]: You might also want to check that database.add is NOT called

**Review by LorenzoBettini — CHANGES_REQUESTED** [review]

**Comment by LorenzoBettini** [comment]
You also need to merge with master, since in the meantime another PR was merged and your PR cannot be merged: fetch from master, merge your branch with master and resolve conflicts, push your branch again (after you addressed all my other comments)

**Comment by Michedev** [comment]
Sorry i read now about the conflicts and now i solved them

**Comment by Michedev** [comment]
Test coverage is decreased because now DbMongo implemenets also the "add" method created by me for the issue

**Review by LorenzoBettini — APPROVED** [review]

**Pull request merged**